core/node/libp2p/rcmgr_logging.go

```diff
@@ -118,6 +118,20 @@ func (n *loggingResourceManager) OpenConnection(dir network.Direction, usefd boo
 	return connMgmtScope, err
 }
 
+func (n *loggingResourceManager) OpenConnectionNoIP(dir network.Direction, usefd bool, endpoint ma.Multiaddr) (network.ConnManagementScope, error) {
+	var connMgmtScope network.ConnManagementScope
+	var err error
+
+	if rcmgr, ok := n.delegate.(rcmgrOpenConnectionNoIPer); ok {
+		connMgmtScope, err = rcmgr.OpenConnectionNoIP(dir, usefd, endpoint)
+	} else {
```

Comment:
Shouldn't we do this be as-is and do this assert in  in `func (n *loggingResourceManager) OpenConnection(dir network.Direction, usefd bool, remote ma.Multiaddr)` instead?

Reply:
I don't think so. Otherwise all normal `OpenConnection` will end up in `OpenConnectionNoIP`.

Reply:
And this fixes the test (locally).